tools/admin/src/main/scala/whisk/core/database/UserCommand.scala

```diff
@@ -171,22 +169,26 @@ class UserCommand extends Subcommand("user") with WhiskCommand {
 
   def createUser(authStore: AuthStore)(implicit transid: TransactionId,
                                        ec: ExecutionContext): Future[Either[CommandError, String]] = {
-    authStore.get[ExtendedAuth](DocInfo(create.subject())).flatMap { auth =>
-      if (auth.isBlocked) {
-        Future.successful(Left(IllegalState(CommandMessages.subjectBlocked)))
-      } else if (auth.namespaces.exists(_.namespace.name == create.desiredNamespace.name)) {
-        Future.successful(Left(IllegalState(CommandMessages.namespaceExists)))
-      } else {
-        val newNS = auth.namespaces + WhiskNamespace(create.desiredNamespace, create.authKey)
-        val newAuth = WhiskAuth(auth.subject, newNS).revision[WhiskAuth](auth.rev)
-        authStore.put(newAuth).map(_ => Right(create.authKey.compact))
+    val authKey = create.auth.map(AuthKey(_)).getOrElse(AuthKey())
+    authStore
+      .get[ExtendedAuth](DocInfo(create.subject()))
+      .flatMap { auth =>
+        if (auth.isBlocked) {
+          Future.successful(Left(IllegalState(CommandMessages.subjectBlocked)))
+        } else if (auth.namespaces.exists(_.namespace.name == create.desiredNamespace(authKey).name)) {
+          Future.successful(Left(IllegalState(CommandMessages.namespaceExists)))
+        } else {
+          val newNS = auth.namespaces + WhiskNamespace(create.desiredNamespace(authKey), authKey)
+          val newAuth = WhiskAuth(auth.subject, newNS).revision[WhiskAuth](auth.rev)
+          authStore.put(newAuth).map(_ => Right(authKey.compact))
+        }
+      }
+      .recoverWith {
+        case _: NoDocumentException =>
+          val auth =
+            WhiskAuth(Subject(create.subject()), Set(WhiskNamespace(create.desiredNamespace(authKey), authKey)))
+          authStore.put(auth).map(_ => Right(authKey.compact))
       }
-    }
-  }.recoverWith {
-    case _: NoDocumentException =>
-      val auth =
-        WhiskAuth(Subject(create.subject()), Set(WhiskNamespace(create.desiredNamespace, create.authKey)))
-      authStore.put(auth).map(_ => Right(create.authKey.compact))
```

Comment:
ha - good one... if `authkey` is a `val` instead of a `def` might have saved you too?

Reply:
Yes. Probably a `lazy val` should have worked as initialization needs to be deferred untill parsing is done